grpc-server/src/main/scala/com/avast/cactus/grpc/server/server.scala

```diff
@@ -1,18 +1,19 @@
 package com.avast.cactus.grpc
 
 import io.grpc._
+import monix.execution.Scheduler
 
 import scala.collection.JavaConverters._
-import scala.concurrent.ExecutionContext
-import scala.reflect.ClassTag
 import scala.language.experimental.macros
+import scala.language.higherKinds
+import scala.reflect.ClassTag
 
 package object server {
 
-  implicit class MapService[ServerTrait](val myTrait: ServerTrait) extends AnyVal {
-    def mappedToService[Service <: BindableService](
-        interceptors: ServerAsyncInterceptor*)(implicit ct: ClassTag[ServerTrait], ec: ExecutionContext): GrpcService[Service] =
-      macro com.avast.cactus.grpc.server.ServerMacros.mapImplToService[Service]
+  implicit class MapService[ServerTrait, F[_]](val myTrait: ServerTrait) extends AnyVal {
+    def mappedToService[Service <: BindableService](interceptors: ServerAsyncInterceptor*)(implicit ct: ClassTag[ServerTrait],
+                                                                                           sch: Scheduler): MappedGrpcService[Service] =
```

Comment:
It's strange for me that you are abstracting using `F[_]` but then you require a `Scheduler` that is specific.

Reply:
I understand. That's unfortunate however I don't know a better solution. Regardless which `F` you choose the `Task` is used internally and it needs the `Scheduler` instance. Maybe @jakubjanecek could help?  
BTW this is common to all our finally-tagless libraries.